nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/datasetbrowser/DatasetBrowserTabController.java

```diff
@@ -0,0 +1,78 @@
+package org.nmrfx.analyst.gui.datasetbrowser;
+
+import javafx.scene.control.Button;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TextField;
+import javafx.scene.control.ToolBar;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+
+
+public abstract class DatasetBrowserTabController {
+    private final Tab tab;
+    protected final BorderPane borderPane = new BorderPane();
+    protected final VBox vBox = new VBox();
+    protected final ToolBar toolBar = new ToolBar();
+    protected final HBox hBox = new HBox();
+    protected final TextField directoryTextField = new TextField();
+    protected DatasetBrowserTableView tableView;
+    protected Button datasetButton;
+    protected Button fetchButton;
+
+    protected DatasetBrowserTabController(String tabName) {
+        tab = new Tab(tabName, borderPane);
+        hBox.getChildren().add(directoryTextField);
+        vBox.getChildren().addAll(toolBar, hBox);
+        borderPane.setTop(vBox);
+        borderPane.setCenter(tableView);
+        HBox.setHgrow(directoryTextField, Priority.ALWAYS);
+        directoryTextField.setOnKeyReleased(e -> {
+            if (e.getCode() == KeyCode.ENTER) {
+                updatePreferences();
+            }
+        });
+
+    }
+
+    protected void initToolbar(boolean includeFetch) {
+        Button retrieveIndexButton = new Button("Index");
+        retrieveIndexButton.setOnAction(e -> retrieveIndex());
+        Button fidButton = new Button("FID");
+        fidButton.setOnAction(e -> openFile(true));
+        datasetButton = new Button("Dataset");
+        datasetButton.setOnAction(e -> openFile(false));
```

Comment:
renamed these methods from fetchDataset to openFile because its confusing they were not actually associated with the "Fetch" button. The "Fetch" button I feel like could also be confusing to the user since it involves changing the "Cached" column of the table,

Reply:
I think the Fetch button should also only be in the remote tab, so it's a bit strange to have it in the base class and  control its addition from a boolean.